src/sonic-config-engine/portconfig.py

```diff
@@ -172,85 +172,152 @@ def parse_port_config_file(port_config_file):
-                if speed_pat is None:
-                    raise Exception('{} speed is not Supported...'.format(speed))
-                speed_G, speed_orig = speed_pat.group(2), speed_pat.group(1)
-                if speed_G:
-                    conv_speed = int(speed_G)*1000
-                else:
-                    conv_speed = int(speed_orig)
-                ports[intf_name]['speed'] = str(conv_speed)
+class BreakoutCfg(object):
+
+    class BreakoutModeEntry:
+        def __init__(self, num_ports, default_speed, supported_speed, num_assigned_lanes=None):
+            self.num_ports = int(num_ports)
+            self.default_speed = self._speed_to_int(default_speed)
+            self.supported_speed = set((self.default_speed, ))
+            self._parse_supported_speed(supported_speed)
+            self.num_assigned_lanes = self._parse_num_assigned_lanes(num_assigned_lanes)
+
+        @classmethod
+        def _speed_to_int(cls, speed):
+            try:
+                if speed.endswith('G'):
+                    return int(speed.replace('G', '')) * 1000
+
+                return int(speed)
+            except ValueError:
+                raise RuntimeError("Unsupported speed format '{}'".format(speed))
+
+        def _parse_supported_speed(self, speed):
+            if not speed:
+                return
+
+            if not speed.startswith('[') and not speed.endswith(']'):
+                raise RuntimeError("Unsupported port breakout format!")
+
+            for s in speed[1:-1].split(','):
+                self.supported_speed.add(self._speed_to_int(s.strip()))
+
+        def _parse_num_assigned_lanes(self, num_assigned_lanes):
+            if not num_assigned_lanes:
+                return
+
+            if isinstance(num_assigned_lanes, int):
+                return num_assigned_lanes
+
+            if not num_assigned_lanes.startswith('(') and not num_assigned_lanes.endswith(')'):
+                raise RuntimeError("Unsupported port breakout format!")
+
+            return int(num_assigned_lanes[1:-1])
+
+        def __eq__(self, other):
+            if isinstance(other, BreakoutCfg.BreakoutModeEntry):
+                if self.num_ports != other.num_ports:
+                    return False
+                if self.supported_speed != other.supported_speed:
+                    return False
+                if self.num_assigned_lanes != other.num_assigned_lanes:
+                    return False
+                return True
             else:
-                raise Exception('Regex return for speed is None...')
+                return False
 
-            ports[intf_name]['index'] = index.split(",")[alias_position]
+        def __ne__(self, other):
+            return not self == other
 
-            parent_intf_id += step
-            alias_position += 1
-            lanes_start += step
+        def __hash__(self):
+            return hash((self.num_ports, tuple(self.supported_speed), self.num_assigned_lanes))
```

Comment:
what exactly are we checking here?

Reply:
BreakoutModeEntry objects are used to compare breakout modes based on their capabilities with __eq__ and __ne__ operators. 
In Python when the class implements __eq__ operator it is recommended to implement __hash__ operator as well. __hash__ operator returns integer hash id of the object that can be used to put an object in a dictionary (as a key) or set. Here hash id is calculated based on capabilities that define breakout mode.

My initial implementation didn't include this method implementation. I added to fix LGTM alerts:

```
This pull request introduces 2 alerts when merging 04be47c into fc91dae - view on LGTM.com

new alerts:

1 for Inconsistent equality and inequality
1 for Inconsistent equality and hashing
```

Reply:
Thanks for the clarification, @oleksandrivantsiv.